### DISCUSSION TITLE
Upgrading from old dsc alarm version

### DISCUSSION THREAD

**Discussion by coolstuff99** [Dilbert66/esphome-dsckeybus#114] [Q&A] [answered]
Hi community,

I have been using dsc alarm from @Dilbert66 for so long but old version (not sure which one but had to download manually old files and add them in home assistant manually).

I proceed to update to the new one which pulls info directly from github. However sensors seems not working as before.
In my previous version, I had each sensor clear/detected individually and now they are permanent "closed".
There is only one entity "DSC Alarm zone status (zs)" that returns changes once detection: OP:1, OP:2, OP:3 and so on...

Is this normal behaviour from now on? is there a way to continue having individual sensors in home assistant providing status changes when clear / detected?

- Always remain closed/clear, no way to get this updated:
![imagen](https://github.com/Dilbert66/esphome-dsckeybus/assets/77806850/83be534c-1717-437d-9689-3cad73469386)

- This one change from ready to stay zones open, same behaviour than before:
![imagen](https://github.com/Dilbert66/esphome-dsckeybus/assets/77806850/d7d971bb-1ece-4fb1-9abf-4e262399716d)

- This one looks new entity detecting one or several zones open but with hardcoded identified OP:X
![imagen](https://github.com/Dilbert66/esphome-dsckeybus/assets/77806850/0026d4c1-6c46-4805-bb86-b811464d5553)

Am I doing something wrong?
Using PC1864 one

Thank you,

**Comment by coolstuff99** [comment, marked as answer]
Fixed!
When comparing both yml files old and recent one, I did a mistake using "id" instead of "id_code" so was not working. Now it works as expected.
Thanks for amazing project!

**Comment by Dilbert66** [comment]
Yes, the id_code was added to allow the system to identify the use of the sensor.  The id field unfortunately is not captured internally so can't be used for that purpose.
